src/test/java/se/sundsvall/paratransit/integration/casedata/mapper/CaseDataMapperTest.java

```diff
@@ -0,0 +1,334 @@
+			.containsExactly(
+				externalCaseId,
+				phase,
+				emptyList(),
+				List.of(
+					new ExtraParameter(keyOther).addValuesItem(valueOther),
+					new ExtraParameter(keyPhaseStatus).addValuesItem(phaseStatus),
+					new ExtraParameter(keyPhaseAction).addValuesItem(phaseAction),
+					new ExtraParameter(keyDisplayPhase).addValuesItem(displayPhase)));
+	}
+
+	@Test
+	void toPatchErrandWithoutDisplayPhase() {
+		final var externalCaseId = "externalCaseId";
+		final var phase = "phase";
+		final var phaseStatus = "phaseStatus";
+		final var phaseAction = "phaseAction";
+		final var keyPhaseAction = "process.phaseAction";
+		final var extraParameters = List.of(new ExtraParameter("key").addValuesItem("value"), new ExtraParameter(keyPhaseAction).addValuesItem("existingPhaseAction"));
+
+		final var bean = CaseDataMapper.toPatchErrand(externalCaseId, phase, phaseStatus, phaseAction, extraParameters);
+
+		assertThat(bean).isNotNull()
+			.hasAllNullFieldsOrPropertiesExcept("externalCaseId", "phase", "extraParameters", "relatesTo", "labels", "facilities")
+			.extracting(
+				PatchErrand::getExternalCaseId,
+				PatchErrand::getPhase,
+				PatchErrand::getFacilities,
+				PatchErrand::getExtraParameters)
+			.containsExactly(
+				externalCaseId,
+				phase,
+				emptyList(),
+				List.of(new ExtraParameter("key").addValuesItem("value"),
+					new ExtraParameter("process.phaseStatus").addValuesItem(phaseStatus),
+					new ExtraParameter(keyPhaseAction).addValuesItem(phaseAction)));
+	}
+
+	@Test
+	void toLawWithNullAsParameters() {
+		final var bean = CaseDataMapper.toLaw(null, null, null, null);
+
+		assertThat(bean).isNotNull().hasAllNullFieldsOrProperties();
+	}
+
+	@Test
+	void toLaw() {
+		final var heading = "heading";
+		final var sfs = "sfs";
+		final var chapter = "chapter";
+		final var article = "article";
+
+		final var bean = CaseDataMapper.toLaw(heading, sfs, chapter, article);
+
+		assertThat(bean).isNotNull()
+			.extracting(
+				Law::getArticle,
+				Law::getChapter,
+				Law::getHeading,
+				Law::getSfs)
+			.containsExactly(
+				article,
+				chapter,
+				heading,
+				sfs);
+	}
+
+	@Test
+	void toAttachmentWithNullAsParameters() {
+		final var bean = CaseDataMapper.toAttachment(null, null, null, null, null);
+
+		assertThat(bean).isNotNull().hasAllNullFieldsOrPropertiesExcept("extraParameters")
+			.extracting(Attachment::getExtraParameters)
+			.isEqualTo(emptyMap());
+	}
+
+	@Test
+	void toAttachment() {
+		final var category = "POLICE_REPORT";
+		final var name = "name";
+		final var extension = "extension";
+		final var mimeType = "mimeType";
+		final var output = "ZmlsZW91dHB1dCBhcyBiYXNlNjQgc3RyaW5n";
+		final var renderedContent = new RenderResponse().output(output);
+
+		final var bean = CaseDataMapper.toAttachment(category, name, extension, mimeType, renderedContent);
+
+		assertThat(bean).isNotNull().hasAllNullFieldsOrPropertiesExcept("category", "name", "extension", "mimeType", "file", "extraParameters")
```

Comment:
Would it not make sense to flip this to `hasNoNullFieldsOrProperties`, feels like youre expecting every field to not be null?

Reply:
Most of the attributes are null in this case.